### PR TITLE
Optimize github actions 

### PR DESCRIPTION
**Pull request by kdorheim**
**Describe development**

Optimize the github actions to minimize the # of action minutes by switching from running tests on mac platform to a Linux platform. 

**Type of development** 

* PATCH: minor changes does not affect Hector code. 

**Checklist**
~- [] [Roxygen](https://roxygen2.r-lib.org/) where applicable but also thorough and coherent inline documentation~
~- [] [Unit tests](https://r-pkgs.org/tests.html) when applicable~
~- [] Complies with the [Hector code style guide](https://jgcri.github.io/hector/articles/manual/StyleGuide.html)~
- [ ] The PR passes automated testing suite

### CODE REVIEW

**Comment by bpbond** [comment]
Oh wow that's an obvious one to change! Thank you @kdorheim

**Review by bpbond — APPROVED** [review]

File: .github/workflows/rcmd.yml [hunk 1]
```diff
@@ -27,12 +27,10 @@ jobs:
       fail-fast: false
       matrix:
         config:
-          # for latest stable version of R for MacOSX
-          - {os: macOS-latest, r: 'release'}
+          # test on one version of R for MacOSX
           - {os: macOS-latest, r: '3.6'}
```
Comment (bpbond) [inline .github/workflows/rcmd.yml:31]: I feel like if we're going to only test on one version, it should be the 'release', not '3.6'?

Comment (kdorheim) [reply .github/workflows/rcmd.yml:31]: 👌

**Comment by bpbond** [comment]
@kdorheim The problem was that R's `curl` package couldn't be installed because there was no underlying curl library available (something about Ubuntu versus Mac defaults I guess). This has been fixed in 721429fc4 by installing it before proceeding with the R installation.

**Comment by kdorheim** [comment]
@bpbond thanks for addressing the issue with the lintr test!

**Pull request merged**